File: 724-find-pivot-index/724-find-pivot-index.py
class Solution(object):
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # Solution: O(n) time, and O(1) additional space.
        
        S = sum(nums)
        left_sum = 0
        
        for i, x in enumerate(nums):
            
            if left_sum == (S - left_sum - x): # Check if left- and right-ward sums are the same.
                return i
            
            left_sum += x # Otherwise, simply increase the left-ward sum.
            
        return -1 # No pivot index found.
        

File: 724-find-pivot-index/724-find-pivot-index.java
class Solution {
    
    public int pivotIndex(int[] nums) {
        
        // Solution: O(n) time, and O(1) additional space.
        int sum = 0, leftSum = 0;
        
        for (int x: nums) sum += x; // Calcuating sum of all elements.
        
        for (int i = 0; i < nums.length; ++i) {
            
            if (leftSum == sum - leftSum - nums[i]) return i; // Check if left- and right-ward sums are the same.
            leftSum += nums[i]; // Otherwise, simply increase the left-ward sum.
            
        }
        
        return -1;
        
    }
    
}